test/with-symlink.test.ts

```diff
@@ -1,52 +1,97 @@
 import * as tap from 'tap';
-const test = tap.test;
 import * as path from 'path';
-
+import * as os from 'os';
+import {exec} from 'child_process'
 import * as subProcess from '../lib/sub-process';
-
 import {jsonParse} from '../lib';
+const test = tap.test;
+
+interface ExecPromiseResponse{
+    stdout: string;
+    stderr: string;
+}
+
+// Since we support Node 8+, we can't use util.promisify(Node 10+). Let's implement promisified exec:
+function execPromise(command:string):Promise<ExecPromiseResponse> {
+  return new Promise(function(resolve, reject) {
+    exec(command, (error, stdout, stderr) => {
+      if (error) {
+        reject(error);
+        return;
+      }
 
-test('with nested GOPATH/src/proj symlink-ing to ../..', (t) => {
+      resolve({stdout, stderr});
+    });
+  });
+}
+
+const isRunningOnWindows = os.platform() === 'win32';
+
+
+test('with nested GOPATH/src/proj symlink-ing to ../..', async(t) => {
   const rootFolder = path.resolve(
     __dirname, 'fixtures', 'proj-with-nested-gopath-and-symlink');
   const gopath = path.resolve(rootFolder, 'gopath');
   const cwd = path.join(gopath, 'src', 'proj');
 
   const manualScriptPath = path.resolve(__dirname, 'manual.ts');
 
+  const nativeExport = isRunningOnWindows ? 'SET' : 'export';
+  if (isRunningOnWindows) {
+      // If we run on windows we have to manually create the symlink every time.
+      // Note that windows symlink generates a file, so we have to delete the current `proj` unix symlink
+    const windowsScriptFile = path.resolve(rootFolder, 'windows-script', 'mklink.cmd');
+    try{
+        // We pass two args: [symlink target location, symlink target location]
+        const msCmd = `${windowsScriptFile} ${cwd} ${rootFolder}`;
```

Comment:
We have to manually create ms symlink (and not syslink like in unix where it is a part of the file data thus commitable)